src/content_mapblock.cpp

```diff
@@ -1165,148 +1165,123 @@ void mapblock_mesh_generate_special(MeshMakeData *data,
 			v3s16 n2p;
 
 			static const v3s16 dirs[6] = {
-				v3s16( 0, 1, 0),
-				v3s16( 0,-1, 0),
-				v3s16( 1, 0, 0),
-				v3s16(-1, 0, 0),
-				v3s16( 0, 0, 1),
-				v3s16( 0, 0,-1)
+				v3s16( 0,  1,  0),
+				v3s16( 0, -1,  0),
+				v3s16( 1,  0,  0),
+				v3s16(-1,  0,  0),
+				v3s16( 0,  0,  1),
+				v3s16( 0,  0, -1)
 			};
 
-			int doDraw[6] = {0,0,0,0,0,0};
+			int doDraw[6] = {0, 0, 0, 0, 0, 0};
 
 			bool drawAllFaces = true;
 
-			bool drawBottomFacesOnly = false; // Currently unused
-
 			// Check for adjacent nodes
-			for(int i = 0; i < 6; i++)
-			{
+			for (int i = 0; i < 6; i++) {
 				n2p = blockpos_nodes + p + dirs[i];
 				n2 = data->m_vmanip.getNodeNoEx(n2p);
 				n2c = n2.getContent();
 				if (n2c != CONTENT_IGNORE && n2c != CONTENT_AIR && n2c != current) {
 					doDraw[i] = 1;
-					if(drawAllFaces)
+					if (drawAllFaces)
 						drawAllFaces = false;
 
 				}
 			}
 
-			for(int j = 0; j < 6; j++)
-			{
-				int vOffset = 0; // Vertical offset of faces after rotation
-				int hOffset = 4; // Horizontal offset of faces to reach the edge
+			for (int j = 0; j < 6; j++) {
+				//float vOffset = 0; // Vertical offset of side faces after rotation
+				float hOffset = 4.0; // Horizontal offset of side faces to reach edge
+				float hOffsetBot = hOffset + 0.7; // Horizontal offset of bottom faces
 
-				video::S3DVertex vertices[4] =
-				{
-					video::S3DVertex(-s,-BS/2,      0, 0,0,0, c, 0,1),
-					video::S3DVertex( s,-BS/2,      0, 0,0,0, c, 1,1),
-					video::S3DVertex( s,-BS/2 + s*2,0, 0,0,0, c, 1,0),
-					video::S3DVertex(-s,-BS/2 + s*2,0, 0,0,0, c, 0,0),
+				video::S3DVertex vertices[4] = {
+					video::S3DVertex(-s, -BS / 2,         0, 0, 0, 0, c, 0, 1),
+					video::S3DVertex( s, -BS / 2,         0, 0, 0, 0, c, 1, 1),
+					video::S3DVertex( s, -BS / 2 + s * 2, 0, 0, 0, 0, c, 1, 0),
+					video::S3DVertex(-s, -BS / 2 + s * 2, 0, 0, 0, 0, c, 0, 0),
 				};
 
 				// Calculate which faces should be drawn, (top or sides)
-				if(j == 0 && (drawAllFaces || (doDraw[3] == 1 || doDraw[1] == 1)))
-				{
-					for(int i = 0; i < 4; i++) {
-						vertices[i].Pos.rotateXZBy(90 + n.param2 * 2);
+				if (j == 0 && (drawAllFaces ||
+						(doDraw[3] == 1 || doDraw[1] == 1))) {
+					for (int i = 0; i < 4; i++) {
+						vertices[i].Pos.rotateXZBy(90);
 						vertices[i].Pos.rotateXYBy(-10);
-						vertices[i].Pos.Y -= vOffset;
+						//vertices[i].Pos.Y -= vOffset;
```

Comment:
why did you comment this out, and not removed completely? (happens at 3 other places as well).